### ISSUE TITLE
Add detachable PTY sessions with detach/reattach support

### ISSUE THREAD

**Issue by zackees** [zackees/running-process#128]
## Goal

Add first-class detachable PTY sessions to `running-process-core`.

The target use case is a client such as `clud` launching an interactive backend in a PTY, detaching quickly, and later reattaching so the session appears suspended/resumed from the user's point of view. This should be a general-purpose `running-process-core` capability with strong cross-platform coverage.

## Desired behavior

- Launch an interactive child behind a PTY from the start.
- Keep the PTY session alive when the foreground client detaches.
- Allow a later client to reattach to the same running session.
- Preserve high-fidelity terminal I/O for TUIs.
- Make detach fast and deterministic.
- Keep detach separate from terminate/kill.
- Support Windows, macOS, and Linux.

## Design direction

Introduce a durable PTY owner/broker abstraction in `running-process-core`.

The foreground process should become an attach client, not the sole owner of the PTY. The broker keeps the PTY master / ConPTY / session resources alive across attach/detach. Attach clients should forward raw terminal bytes so terminal queries, escape sequences, paste, resize, and TUI input continue to work correctly.

Reattach can use backlog replay, terminal-state capture/repaint, or another tested strategy to restore the visible screen.

## Possible API sketch

```rust
let session = NativePtySession::spawn(config)?;
let attach = session.attach()?;
attach.run_foreground()?;
attach.detach()?;

let session = NativePtySession::open(session_id)?;
session.attach()?.run_foreground()?;
session.terminate_tree()?;
```

## Testing plan

This should be built experimentally and tested thoroughly across supported platforms:

- Cross-platform integration tests on Windows, macOS, and Linux.
- Child remains alive after detach.
- Reattach restores interactive input/output.
- Ctrl-C detach is fast and does not kill the child.
- Explicit terminate kills the full child tree.
- Broker/owner death cleans up correctly.
- Resize propagates before and after reattach.
- Raw terminal byte forwarding handles escape replies, paste, arrows, control keys, and TUI startup queries.
- Reattach behavior is validated with real PTY-backed processes, not only mocked process handles.

## Context

This came out of experimenting with `clud` daemon-managed interactive sessions. Plain inherited-stdio launches provide excellent terminal fidelity, while daemon-owned PTY sessions make detach/reattach possible. The right long-term direction is to make detachable PTYs a reliable primitive in `running-process-core`, then let clients like `clud` build suspend/resume behavior on top of that.

**Comment by zackees** [comment]
Closing this as completed by the #130 implementation series. #128 was the original detachable PTY feature ask; #130 expanded it into the full PTY + pipe session scenario matrix and is now closed.

Relevant merged PRs include:

- #133 / #134 / #135: detachable PTY session RPC scaffold, implementation, and cross-process attach test.
- #136: pipe-backed session parity.
- #137, #143, #146, #147, #149: fast Ctrl+C / soft-then-hard termination and outcome coverage.
- #138, #139, #142: sessions list/log/terminate/purge/kill CLI and backlog query support.
- #140: session cleanup on daemon shutdown.
- #141, #144, #145, #148: detached backlog replay, resize while detached, non-TTY attach mode, and daemon autostart sessions.

The current `main` contains daemon-owned PTY sessions, detach/reattach, backlog replay, resize, single-attach/concurrent attach behavior, pipe parity, termination, and cross-process tests. That satisfies this original #128 scope.
